### PR TITLE
DIP0022: Multi-Quorum ChainLocks

### PR DESCRIPTION
**Pull request by UdjinM6**
Alternative to #85

### CODE REVIEW

**Comment by DarrenTa** [comment]
I think this option provides a much cleaner approach to the documentation.  

I'm still reviewing this and comparing it to 0008.  It'll take me having a day without distractions which I have later this week.

**Review by xdustinface — CHANGES_REQUESTED** [review]
I like the separation 👍 Im not sure about the calculation section though.. I think we shouldn't just partially copy it over to here since you anyway need to open DIP8 because of the missing context if you want to understand it and then you can just look there (i.e. the `N` and `m` of the table are not even mentioned in here). Would probably make more sense to just write few words and link to https://github.com/dashpay/dips/blob/master/dip-0008.md#calculations there or so, thoughts?

Also see some typos and two potential generalizations in comments below.

**Comment by UdjinM6** [comment]
Rebased to fix merge conflicts after #87

**Comment by DarrenTa** [comment]
OK explicit answer to @UdjinM6's question on the other thread

>Should be accurate for a single LLMQ mode but yeah, for a multi-quorum mode you'd need to disrupt 2 quorums (or control 3 quorums to produce invalid clsig) at the same time, so I guess you should take all these probabilities roughly to the power of 2 (or 3 respectively) or so? Let's ping @DarrenTa :)

Quick answer: cubing is correct, squaring is not so simple.

p and q are probabilities, and are numbers between 0 and 1

So if the chance of an event happening is p then the chances of it happening three times in a row is p^3 assuming the chance of the event happening does not change.

However if the chance of an event happening is p then the chance of it not happening is 1-p=q.  If the experiment happens three times then there is a p*p*q chance that the event happens the first two times but does not happen the third time. The chance that the event happens the first and last time but not the second is p*q*p.  The chance the event does not happen the last two times but not the first is q*p*p.

The result is the chance the event happens **exactly** two times is 3*p^2*q
The chance that the event happens three times is p^3 (these events on these two lines are mutually exclusive)
Thus the chance the event happens **at least** two times is 3*p^2*q + q^3

**Comment by DarrenTa** [comment]
A major assumption in the above comment is that the chance of the event p does not change.  

Depending on how the quorums are picked this may or may not be the case.  If each quorum is a simple random selection then the assumption holds.  

**Alternate selection** If the three quorums are picked by listing all MN in an order and picking the first few for the first quorum the next few for the second quorum and a third few for the third quorum then this is not a simple random sample.

If the attacker has less than 40% of the MNs and using the alternate selection the attacker gets over 40% in the first quorum then the attacker will be less likely to have nodes in the second or third quorum compared to a simple random selection.

File: dip-0022.md [hunk 1]
```diff
@@ -95,8 +96,20 @@ The sign hash for the BLS signature is calculated as
 The request id for each LLMQ is calculated as
 `SHA256("clsig", blockHeight, quorumHash)`.
 
+To ensure the convergence of block signing we reuire honest quorums to form a
```
Comment (thephez) [inline dip-0022.md:99]:
```suggestion
To ensure the convergence of block signing we require honest quorums to form a
```

**Pull request closed without merging**